# Analysis of the *Installed* Dataset
### Question 1: *Merge the filtered rows (from question 1-3) from the Licence dataset with the installed elevators. How many rows are merged from each dataset ? How many rows does the merged dataset have ?*

In [2]:
import pandas as pd
license = pd.read_csv("../data/raw/licence.csv")
installed = pd.read_json("../data/raw/installed.json")
for df in [installed, license]:
    df.columns = df.columns.str.upper()
    df.columns = df.columns.str.replace(' ', '')

activeLicense = license[license["LICENSESTATUS"] == "ACTIVE"]
merged = pd.merge(activeLicense, installed, on='ELEVATINGDEVICESNUMBER')
print('Amount of rows in the merged dataset: ' + str(len(merged))) 

Amount of rows in the merged dataset: 42665


### Conclusion: *We can now see the amount of rows in the merged dataset. If my understanding is correct we got all rows from the* license *dataset and 42665 rows from the* installed *set.*
- - - 

### Question 2: *Both datasets have a variable about the location of the elevator. Show the observations where both variables are not equal. What do you notice ?*

In [8]:
import numpy as np
diff = merged.loc[merged["LOCATIONOFTHEELEVATINGDEVICE"] != merged["LICENSEHOLDERADDRESS"]]
print(str(len(diff) * 100 / len(merged)) + '% of rows are different.')

100.0% of rows are different.


In [12]:
diff
# diff['C'] = [i.replace(j, '').strip('()') for i, j in zip(diff["LOCATIONOFTHEELEVATINGDEVICE"], diff["LICENSEHOLDERADDRESS"])]
# diff['C']=diff["LOCATIONOFTHEELEVATINGDEVICE"].replace(regex=r'\('+ diff["LICENSEHOLDERADDRESS"] +r'\)',value="")

AttributeError: 'float' object has no attribute 'replace'